pkg/util/error_list.go

```diff
@@ -45,10 +45,67 @@ func (list errorList) Error() string {
 	return result
 }
 
+// Except returns any errors (or nested errors, if the list contains
+// nested []error slices) for which all fns return false. If no errors
+// remain a nil list is returned. The resulting list will have all
+// nested []error slices flattened as a side effect.
```

Comment:
Are they?  We don't wrap them with a parent error, so there's nothing grouping them.  To me, an error that is indivisible is a struct { []error } - an aggregate error list is []error.

This code is for the case in the builder where lots of sources (two directory scans) can each generate errors on individual files, and the flattened list of lists is what you would filter for non-relevant errors.


> On Jan 4, 2015, at 1:29 AM, Tim Hockin <notifications@github.com> wrote:
> 
> In pkg/util/error_list.go:
> 
> > @@ -45,10 +45,67 @@ func (list errorList) Error() string {
> >  	return result
> >  }
> >  
> > +// Except returns any errors (or nested errors, if the list contains
> > +// nested []error slices) for which all fns return false. If no errors
> > +// remain a nil list is returned. The resulting list will have all
> > +// nested []error slices flattened as a side effect.
> flattening sub-slices seems wrong to me - They are each logically a single error.
> 
> —
> Reply to this email directly or view it on GitHub.
>